dev-lang/nqp/nqp-2021.10.ebuild

```diff
@@ -0,0 +1,159 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit java-pkg-opt-2 multibuild
+
+if [[ ${PV} == "9999" ]]; then
+	EGIT_REPO_URI="https://github.com/perl6/${PN}.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/perl6/${PN}/releases/download/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Not Quite Perl, a Perl 6 bootstrapping compiler"
+HOMEPAGE="https://rakudo.org"
+
+LICENSE="Artistic-2"
+SLOT="0"
+IUSE="doc clang java +moar test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="|| ( java moar )"
+
+CDEPEND="java? (
+		dev-java/asm:4
+		dev-java/jline:0
+		dev-java/jna:4
+	)
+	moar? ( ~dev-lang/moarvm-${PV}[clang=] )"
+RDEPEND="${CDEPEND}
+	java? ( >=virtual/jre-1.8 )"
+DEPEND="${CDEPEND}"
+BDEPEND="${CDEPEND}
+	clang? ( sys-devel/clang )
+	java? ( >=virtual/jdk-1.8 )
```

Comment:
I think we're supposed to use :* if it works with any Java version as far as we know.

(Also, note that nqp seems to get the BDEPEND part right.)